eyeshade/workers/reports.js

```diff
@@ -109,6 +118,48 @@ const daily = async (debug, runtime) => {
   debug('daily', 'running again ' + moment(tomorrow).fromNow())
 }
 
+exports.freezeOldSurveyors = freezeOldSurveyors
+
+async function freezeOldSurveyors (olderThanDays, anchorTime, surveyors) {
+  // in seconds
+  const nDays = DAY_SECONDS * olderThanDays
+  const nDaysAgo = (anchorTime - nDays)
+  const nDaysNumber = nDaysAgo * 1000
+  const where = {
+    frozen: false,
+    surveyorType: 'contribution'
+  }
+  const data = {
+    $set: {
+      frozen: true
+    }
+  }
+  const allNonFrozenSurveyors = await surveyors.find(where)
+  const freezeUpdates = allNonFrozenSurveyors.map(surveyorExpirer)
+  await Promise.all(freezeUpdates)
+    .then(() => freezeUpdates.length)
+
+  function surveyorExpirer (surveyor) {
+    const {
+      _id,
+      timestamp
+    } = surveyor
+    const where = {
+      _id
+    }
+    if (!timestamp) {
```

Comment:
you can probably ditch this. assuming it doesn't hurt to set frozen on a bad surveyor

Reply:
that's fair